src/se/liu/ida/paperio/AI.dl

```diff
@@ -18,7 +18,16 @@ limiteXMin(0). limiteYMin(0).
 :- player(_, _, N1, _, _), player(_, _, N2, _, _), N1=N2.
 % ---> la distanza tra il player è il nemico viene calcolata dalla distanza tra la tile del player e quella dell'enemy
 % in fase di sperimentazione
-%------- distancePlayer(Player, distanceX, distanceY, Enemy):- currentPlayer(Player),player(X,Y,Player,_,_),player(Ex,Ey,Enemy,_,_),Player != Enemy
+distancePlayer(Player, DistanceX, DistanceY, Enemy):- currentPlayer(Player),player(X,Y,Player,_,_),player(Ex,Ey,Enemy,_,_),Player != Enemy, Px>=Ex, Py >=Ey, DistanceX=Px - Ex, DistanceY = Py - Ey.
```

Comment:
Perché avere `distancePlayer()` con `distanceX` e `distanceY`? Serve avere la distanza come scalare. In più, player può essere omesso. Si potrebbe avere un qualcosa come:
```suggestion
distanceFromPlayer(Distance, Enemey):- currentPlayer(P), player(PX, PY, P, _, _), player(EX, EY, Enemy, _, _), P!=Enemy, Distance = &abs(PX - EX) + &abs(PY - EY).
```